compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -1934,9 +1936,146 @@ public BType getSafeType(BType type, boolean liftError) {
+            case TypeTags.STREAM:
+            case TypeTags.TABLE:
+            case TypeTags.XML:
+            case TypeTags.ANYDATA:
+            case TypeTags.ANY:
+                return true;
+            case TypeTags.ANNOTATION:
+            case TypeTags.CHANNEL:
+            case TypeTags.ERROR:
+            case TypeTags.FUTURE:
+            case TypeTags.INVOKABLE:
+            case TypeTags.PACKAGE:
+            case TypeTags.SEMANTIC_ERROR:
+            case TypeTags.XML_ATTRIBUTES:
+                return false;
+            default:
+                return analyzeImplicitInitialValue(type);
+        }
+    }
+
+    private boolean analyzeImplicitInitialValue(BType type) {
+        switch (type.tag) {
+            case TypeTags.ARRAY:
+                BArrayType arrayType = (BArrayType) type;
+                if (arrayType.state == BArrayState.UNSEALED) {
+                    // Empty array is the implicit initial value for arrays.
+                    return true;
+                }
+                return hasImplicitInitialValue(arrayType.eType);
+            case TypeTags.FINITE:
+                return analyzeFiniteType((BFiniteType) type);
+            case TypeTags.OBJECT:
+                return analyzeObjectType((BObjectType) type);
+            case TypeTags.RECORD:
+                BRecordType recordType = (BRecordType) type;
+                return recordType.fields.stream().allMatch(f -> hasImplicitInitialValue(f.type));
+            case TypeTags.TUPLE:
+                BTupleType tupleType = (BTupleType) type;
+                return tupleType.tupleTypes.stream().allMatch(this::hasImplicitInitialValue);
+            case TypeTags.UNION:
+                return analyzeUnionType((BUnionType) type);
+            default:
+                return false;
+        }
+    }
+
+    private boolean analyzeUnionType(BUnionType type) {
+        // NIL is a member.
+        if (type.memberTypes.stream().anyMatch(t -> t.tag == TypeTags.NIL)) {
+            return true;
+        }
+
+        // Value space contains nil.
+        if (type.memberTypes.stream().anyMatch(t -> t.isNullable())) {
+            return true;
+        }
+
+        // All members are of same type and has the implicit initial value as a member.
+        Iterator<BType> iterator = type.memberTypes.iterator();
+        BType firstMember;
+        for (firstMember = iterator.next(); iterator.hasNext(); ) {
+            if (!isSameType(firstMember, iterator.next())) {
+                return false;
+            }
+        }
+        // Control reaching this point means there is only one type in the union.
+        return isValueType(firstMember) && hasImplicitInitialValue(firstMember);
+    }
+
+    private boolean analyzeObjectType(BObjectType type) {
+        if (type.getKind() == TypeKind.SERVICE) {
+            return false;
+        }
+        if (type.tsymbol instanceof BObjectTypeSymbol) { // To please the FindBugs warning, BC_UNCONFIRMED_CAST
```

Comment:
lets use the symbol kind for the comparison. Thats what we have done elsewhere.

Reply:
IIRC before moving into Types class, travis failed with commented error message. Let me try again with symbol kind.